addons/web/static/src/core/debug/debug_context.js

```diff
@@ -60,17 +60,27 @@ class DebugContext {
     }
 }
 
+let currentDebugContext;
 const debugContextSymbol = Symbol("debugContext");
 export function useOwnDebugContext({ categories = [] } = {}) {
-    useSubEnv({ [debugContextSymbol]: new DebugContext(useEnv(), categories) });
+    const oldDebugContext = currentDebugContext;
+    currentDebugContext = new DebugContext(useEnv(), categories);
+    useSubEnv({ [debugContextSymbol]: currentDebugContext });
+    useEffect(
+        () => {
+            return () => {
+                currentDebugContext = oldDebugContext;
```

Comment:
This is asymmetric since we set the currentDebugContext in the hook body (during component setup) but we restore the old context on unmount, if the component is unmouted then remounted then remounted this will break. We really need an "onDestroyed" hook :grimacing: